src/scripts/init.js

```diff
@@ -0,0 +1,36 @@
+const appName = require('../../package.json').name;
+
+$(document).ready(function () {
+
+	const mountPoint = $('<div>', {
+		id: appName,
+		html: '<div>'
+	});
+
+	if (!OCA.Files) {
+		return;
+	}
+
+	// ---- Register fileactions -------
+
+	let ViewMedia = {
+		mime: 'image/png',
+		name: appName,
+		permissions: OC.PERMISSION_READ,
+		actionHandler(fileName, context) {
+
+			window[appName] = {
+				fileName,
+				context
+			};
+
+			window.location = ['#', appName, fileName].join('/');
+
+			$('body').append(mountPoint);
+			OC.addScript(appName, appName);
```

Comment:
what if the URL changes while the script is loading ?
somehow I thought we'd be using `.then()` here to boot the viewer then and pass it the path

Reply:
the app is only interested in the hash value. It will react to any changes no matter if they happen ahead of time or later on. Still, I moved the redirect into the callback.